src/transformStateWithClones.js

```diff
@@ -8,6 +9,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const table = {
+    ...state,
+  };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      for (const delet in table) {
+        delete table[delet];
+      }
+      console.log(table);
+    } else if (action.type === 'removeProperties') {
+      for (const remove of action.keysToRemove) {
+        delete table[remove];
+      }
+    } else {
+      Object.assign(table, action.extraData);
+    }
+
+    result.push({ ...table });
```

Comment:
It is better to push a copy of the object to the array at the end of the each loop cycle, but outside of the `switch` block, after current action is processed. This comment is a reminder that you should replace the `if...else` construct with a `switch` block and then place this line after the `switch` block.